lib/events/event_handler.ex

```diff
@@ -56,6 +59,27 @@ defmodule ElixirTools.Events.EventHandler do
     :ok
   end
 
+  @spec publish(Event.t(), event_schema, [events_opt]) :: :ok
+  def publish(event, schema, opts) do
+    event_module = opts[:event_module] || Event
+    task_supervisor_module = opts[:task_supervisor_module] || Task.Supervisor
+    not_sent_event_module = opts[:not_sent_event_module] || NotSentEvent
+
+    case event_module.validate_json_schema(schema, event) do
+      :ok ->
+        task_supervisor_module.async_nolink(ElixirTools.TaskSupervisor, fn ->
+          publish_event_call(event, opts)
+        end)
+
+      {:error, reason} ->
```

Comment:
in case of error we also need 
```elixir
        telemetry_module.execute(
          [:pagantis_elixir_tools, :events, :not_sent],
          %{error_info: error_info}
        )
```
probably we need to but 
```elixir
        error_info = event |> Map.from_struct() |> Map.put(:reason, inspect(reason))

        telemetry_module.execute(
          [:pagantis_elixir_tools, :events, :not_sent],
          %{error_info: error_info}
        )

        not_sent_event_module.create!(%{content: Jason.encode!(error_info)})
```
into a dedicated function which will be called for both failed validation and failed send cases